plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php

```diff
@@ -128,6 +149,50 @@ public static function transient_update_themes( $transient ) {
 		return $transient;
 	}
 
+	/**
+	 * Runs on load-plugins.php, adds a hook to show a custom plugin update message for Woo.com hosted plugins.
+	 *
+	 * @return void.
+	 */
+	public static function setup_update_plugins_messages() {
+		foreach ( WC_Helper::get_local_woo_plugins() as $plugin ) {
+			$filename = $plugin['_filename'];
+			add_action( 'in_plugin_update_message-' . $filename, array( __CLASS__, 'add_install_marketplace_plugin_message' ), 10, 2 );
+		}
+	}
+
+	/**
+	 * Runs on in_plugin_update_message-{file-name}, show a message to install the Woo Marketplace plugin, on plugin update notification,
+	 * if the Woo Marketplace plugin isn't already installed.
+	 *
+	 * @param object $plugin_data TAn array of plugin metadata.
+	 * @param object $response  An object of metadata about the available plugin update.
+	 *
+	 * @return void.
+	 */
+	public static function add_install_marketplace_plugin_message( $plugin_data, $response ) {
+		if ( ! empty( $response->package ) || WC_Helper_Plugin::is_plugin_active() ) {
+			return;
+		}
+		if ( ! WC_Helper_Plugin::is_plugin_installed() ) {
+			printf(
+			/* translators: 1: Marketplace plugin install URL 2: Marketplace plugin download URL */
+				wp_kses(
+					' <a href="%1$s">Install Woo Connect to Update</a>.',
+					array(
+						'a' => array(
+							'href' => array(),
+						),
+					)
+				),
+				esc_url( 'https://woo.com/woocom-plugin/install/' ),
+				esc_url( 'https://woo.com/woocom-plugin/download/' ),
+			);
+		} elseif ( ! WC_Helper_Plugin::is_plugin_active() ) {
+			printf( ' Activate Woo Connect to Update.' );
+		}
```

Comment:
For consistency of the code across all method, and better readability, could be better to change to this variant:

```suggestion
		if ( ! WC_Helper_Plugin::is_plugin_installed() ) {
			printf(
			/* translators: 1: Marketplace plugin install URL 2: Marketplace plugin download URL */
				wp_kses(
					' <a href="%1$s">Install Woo Connect to Update</a>.',
					array(
						'a' => array(
							'href' => array(),
						),
					)
				),
				esc_url( 'https://woo.com/woocom-plugin/install/' ),
				esc_url( 'https://woo.com/woocom-plugin/download/' ),
			);
			return;
		} 
		
		if ( ! WC_Helper_Plugin::is_plugin_active() ) {
			printf( ' Activate Woo Connect to Update.' );
			return;
		}
```

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/44279/commits/309112100962962404bd6024ee1851a314502f08